src/riak_kv_wm_utils.erl

```diff
@@ -335,10 +338,71 @@ is_valid_referer(RD) ->
         OriginTuple ->
             true;
         RefererTuple ->
-            lager:debug("WM referrer not origin.  Origin ~p != Referer ~p\n", [OriginTuple, RefererTuple]),
-            false
+            allowable_origin(OriginTuple, RefererTuple)
```

Comment:
This is a minor style concern, but I would find `allowable_origin/2` a bit easier to read if, instead of handling `OriginTuple == OriginTuple` in `is_valid_referer/1` directly, it was a separate function head.

So: is_valid_referer's case statement would be undefined/invalid/Referer, and then the first function head for `allowable_origin/2` would be `(Origin, Origin) -> true`.